packages/langium/test/validation/document-validator.test.ts

```diff
@@ -259,4 +259,82 @@ describe('Register Before/AfterDocument logic for validations with state', () =>
+        `;
+
+        let validate: (input: string, options?: ValidationHelperOptions) => Promise<ValidationResult<AstNode>>;
+
+        beforeAll(async () => {
+            const services = await createServicesForGrammar({
+                grammar
+            });
+            const validationChecksUser: ValidationChecks<object> = {
+                AstNode: [
+                    (node, accept) => {
+                        accept('error', 'TEST', { node });
+                    }
+                ]
+            };
+            services.validation.ValidationRegistry.register(validationChecksUser, services.validation.ValidationRegistry, 'user');
+            const validationChecksFast: ValidationChecks<object> = {
+                AstNode: [
+                    (node, accept) => {
+                        accept('warning', 'TEST', { node });
+                    }
+                ]
+            };
+            services.validation.ValidationRegistry.register(validationChecksFast);
+
+            validate = validationHelper(services);
+        });
+
+        test('empty categories', async () => {
+            const validationResult = await validate('model test', {validation:{categories:[]}});
+            const diagnostics = validationResult.diagnostics;
+            expect(diagnostics).toHaveLength(0);
+            expect(diagnostics.filter(d => d.severity === 1)).toHaveLength(0); // 0 error
+            expect(diagnostics.filter(d => d.severity === 2)).toHaveLength(0); // 0 warning
+        });
+
+        test('fast category', async () => {
+            const validationResult = await validate('model test', {validation:{categories:['fast']}});
+            const diagnostics = validationResult.diagnostics;
+            expect(diagnostics).toHaveLength(1);
+            expect(diagnostics.filter(d => d.severity === 1)).toHaveLength(0); // 0 error
+            expect(diagnostics.filter(d => d.severity === 2)).toHaveLength(1); // 1 warning
+        });
+
+        test('user category', async () => {
+            const validationResult = await validate('model test', {validation:{categories:['user']}});
+            const diagnostics = validationResult.diagnostics;
+            expect(diagnostics).toHaveLength(1);
+            expect(diagnostics.filter(d => d.severity === 1)).toHaveLength(1); // 1 error
+            expect(diagnostics.filter(d => d.severity === 2)).toHaveLength(0); // 0 warning
+        });
+
+        test('user&fast categories', async () => {
+            const validationResult = await validate('model test', {validation:{categories:['user', 'fast']}});
+            const diagnostics = validationResult.diagnostics;
+            expect(diagnostics).toHaveLength(2);
+            expect(diagnostics.filter(d => d.severity === 1)).toHaveLength(1); // 1 error
+            expect(diagnostics.filter(d => d.severity === 2)).toHaveLength(1); // 1 warning
+        });
+
+        test('default categories', async () => {
```

Comment:
I'm surprised that by default all categories are enabled. I was expecting only `built-in` and `fast` checks.

I don't know if the behavior is different between the test framework and the default builder.

Reply:
ok,

In the test framework  `buildOptions.validation` is true by default so all validator are executed.
In the DefaultBuilder `buildOptions.validation` is a ValidationOptions with categories `built-in` and `fast`